slatedb/src/tablestore.rs

```diff
@@ -664,7 +700,13 @@ mod tests {
             min_filter_keys: 1,
```

Comment:
we should add some tests to assert that we use the right object store type for the various operations

Reply:
could you outline what the tests might look like?

Reply:
probably what would make sense is:
1. some tests for writing and reading back a small SST of each type, and listing ssts. The test can be a table test with cases for the same and different object stores.
2. some test that writes an sst of each type and uses the object store directly to make sure the file is in the right place

Reply:
something like this https://github.com/slatedb/slatedb/pull/558#discussion_r2089519200 ?

Reply:
https://github.com/slatedb/slatedb/pull/558/commits/51e10ba8c29a32ff9590e522e4040459fc748856